src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java

```diff
@@ -53,21 +62,30 @@ public static void generateXdocContent() throws Exception {
         for (Path path : templatesFilePaths) {
             final String pathToFile = path.toString();
             final File inputFile = new File(pathToFile);
-            final File outputFile = new File(pathToFile.replace(".template", ""));
+            final File tempFile = File.createTempFile(pathToFile.replace(".template", ""), "");
+            try {
+                final XdocsTemplateSinkFactory sinkFactory = (XdocsTemplateSinkFactory)
+                       plexus.lookup(SinkFactory.ROLE, XDOCS_TEMPLATE_HINT);
+                final Sink sink = sinkFactory.createSink(tempFile.getParentFile(),
+                       tempFile.getName(), String.valueOf(StandardCharsets.UTF_8));
+                final XdocsTemplateParser parser = (XdocsTemplateParser)
+                       plexus.lookup(Parser.ROLE, XDOCS_TEMPLATE_HINT);
+                try (Reader reader = ReaderFactory.newReader(inputFile,
+                       String.valueOf(StandardCharsets.UTF_8))) {
+                    parser.parse(reader, sink);
+                }
+                final File outputFile = new File(pathToFile.replace(".template", ""));
+
+                final Object exp = StandardCopyOption.REPLACE_EXISTING;
 
-            final XdocsTemplateSinkFactory sinkFactory = (XdocsTemplateSinkFactory)
-                    plexus.lookup(SinkFactory.ROLE, XDOCS_TEMPLATE_HINT);
-            final Sink sink = sinkFactory.createSink(outputFile.getParentFile(),
-                    outputFile.getName(), String.valueOf(StandardCharsets.UTF_8));
-            final XdocsTemplateParser parser = (XdocsTemplateParser)
-                    plexus.lookup(Parser.ROLE, XDOCS_TEMPLATE_HINT);
-            try (Reader reader = ReaderFactory.newReader(inputFile,
-                    String.valueOf(StandardCharsets.UTF_8))) {
-                parser.parse(reader, sink);
+                Files.copy(tempFile.toPath(), outputFile.toPath(), (CopyOption) exp);
+
+                tempFile.delete();
             }
-            finally {
-                sink.close();
+            catch (IOException ex) {
```

Comment:
We need to close sink

Reply:
Done @romani

Reply:
Do not catch exception, let fly all the way up.
Update method signature to allow it.

Reply:
@romani Do you mean this way ?
```
  catch (IOException ex) {
                
            }
```

Reply:
@romani I cannot remove the catch block it is causing me an error.

Reply:
Share error

Reply:
Code: 
f```
inal PlexusContainer plexus = new DefaultPlexusContainer();
        final Set<Path> templatesFilePaths = XdocUtil.getXdocsTemplatesFilePaths();

        for (Path path : templatesFilePaths) {
            final String pathToFile = path.toString();
            final File inputFile = new File(pathToFile);
            final File tempFile = File.createTempFile(pathToFile.replace(".template", ""), "");
            try {
                final XdocsTemplateSinkFactory sinkFactory = (XdocsTemplateSinkFactory)
                        plexus.lookup(SinkFactory.ROLE, XDOCS_TEMPLATE_HINT);
                final Sink sink = sinkFactory.createSink(tempFile.getParentFile(),
                        tempFile.getName(), String.valueOf(StandardCharsets.UTF_8));
                final XdocsTemplateParser parser = (XdocsTemplateParser)
                        plexus.lookup(Parser.ROLE, XDOCS_TEMPLATE_HINT);
                try (Reader reader = ReaderFactory.newReader(inputFile,
                        String.valueOf(StandardCharsets.UTF_8))) {
                    parser.parse(reader, sink);
                }
                final File outputFile = new File(pathToFile.replace(".template", ""));

                final Object exp = StandardCopyOption.REPLACE_EXISTING;

                Files.copy(tempFile.toPath(), outputFile.toPath(), (CopyOption) exp);

                tempFile.delete();

                sink.close();
            }


        }
```
**Error Image without catch block** : 
![Uploading Screenshot 2023-10-13 at 5.36.40 AM.png…]()

Reply:
You need to do finally block for nested TRY, and close sink there

Reply:
@romani Still need catch/block for the outer try: 
<img width="2056" alt="Screenshot 2023-10-13 at 6 14 20 AM" src="https://github.com/checkstyle/checkstyle/assets/60972989/77acb6b2-bfd6-4250-9ea3-49b5f04f557a">

Reply:
Can we avoid outer catch ?

Reply:
Done